core/server/utils/startup-check.js

```diff
@@ -9,11 +9,52 @@ var packages = require('../../../package.json'),
 
 checks = {
     check: function check() {
+        this.nodeVersion();
+        this.nodeEnv();
         this.packages();
         this.contentPath();
         this.sqlite();
     },
 
+    // Make sure the node version is supported
+    nodeVersion: function checkNodeVersion() {
+        // Tell users if their node version is not supported, and exit
+        try {
+            var semver = require('semver');  
+        } catch (e) {
+            return;
+        }   
+        if (!semver.satisfies(process.versions.node, packages.engines.node) &&
+            !semver.satisfies(process.versions.node, packages.engines.iojs)) {
+            console.error('\x1B[31mERROR: Unsupported version of Node');
+            console.error('\x1B[31mGhost needs Node version ' + packages.engines.node +
+                            ' you are using version ' + process.versions.node + '\033[0m\n');
+            console.error('\x1B[32mPlease go to http://nodejs.org to get a supported version\033[0m');
+            
+            process.exit(0);
+        }
+    },
+
+    nodeEnv: function checkNodeEnvState() {
+        // Check if config path resolves, if not check for NODE_ENV in config.example.js prior to copy
+        try {
+            fd = fs.openSync(configFilePath, 'r');
```

Comment:
This is going to fail linting.

`fd` is not declared, and below you have two `var` statements.  It's going to have to be adjusted to top declare the vars you need.